### PR TITLE
fix: invalid links in markdown files

### PR DESCRIPTION
**Pull request by juzhiyuan**
### What this PR does / why we need it:

fix invalid links in markdown files, not only README

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: docs/zh/latest/FAQ.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ APISIX 是当前性能最好的 API 网关，单核 QPS 达到 2.3 万，平均
 4. 变化通知
 5. 高性能
 
-APISIX 需要一个配置中心，上面提到的很多功能是传统关系型数据库和KV数据库是无法提供的。与 etcd 同类软件还有 Consul、ZooKeeper等，更详细比较可以参考这里：[etcd why](https://github.com/etcd-io/etcd/blob/master/Documentation/learning/why.md#comparison-chart)，在将来也许会支持其他配置存储方案。
+APISIX 需要一个配置中心，上面提到的很多功能是传统关系型数据库和 KV 数据库是无法提供的。与 etcd 同类软件还有 Consul、ZooKeeper 等，更详细比较可以参考这里：[etcd why](https://etcd.io/docs/v3.3.12/learning/why/)，在将来也许会支持其他配置存储方案。
```
Comment (spacewander) [inline docs/zh/latest/FAQ.md:61]: We don't use v3.3

File: docs/zh/latest/admin-api.md [hunk 2]
```diff
@@ -23,75 +23,76 @@ title: Admin API
...
+- [SSL](#ssl)
+- [Global Rule](#global-rule)
+- [Plugin Config](#plugin-config)
+- [Plugin Metadata](#plugin-metadata)
+- [Plugin](#plugin)
 
 ## Route
 
-*地址*：/apisix/admin/routes/{id}?ttl=0
+_地址_：/apisix/admin/routes/{id}?ttl=0
```
Comment (spacewander) [inline docs/zh/latest/admin-api.md:39]: I am confusing by the style. Why sometimes you change `*blah*` to `_blah_` but sometimes you change `_blah_` to `*blah*`? And why should we change the style?

File: docs/en/latest/plugins/hmac-auth.md [hunk 3]
```diff
@@ -101,24 +105,24 @@ The calculation formula of the signature is `signature = HMAC-SHAx-HEX(secret_ke
 
 > The coding steps of canonical_query_string are as follows:
 
-* Extract the `query` item in the URL, that is, the string "key1 = valve1 & key2 = valve2" after the "?" in the URL.
```
Comment (spacewander) [inline docs/en/latest/plugins/hmac-auth.md:104]: Why change this part?

**Review by spacewander — CHANGES_REQUESTED** [review]

File: docs/en/latest/plugins/hmac-auth.md (hunk 3 above)
Comment (juzhiyuan) [reply docs/en/latest/plugins/hmac-auth.md:104]: All changes except broken links happened due to VSCode auto-format 🤔

File: docs/zh/latest/FAQ.md (hunk 1 above)
Comment (juzhiyuan) [reply docs/zh/latest/FAQ.md:61]: Please provide a correct link.

Comment (Yiyiyimu) [reply docs/zh/latest/FAQ.md:61]: You could try `https://etcd.io/docs/v3.4.0/learning/why/` or `https://etcd.io/docs/current/learning/why/`

File: docs/zh/latest/plugin-develop.md [hunk 4]
```diff
@@ -299,9 +303,9 @@ end
 
 ## 编写测试用例
 
-针对功能，完善各种维度的测试用例，对插件做个全方位的测试吧！插件的测试用例，都在 __t/plugin__ 目录下，可以前去了解。
-项目测试框架采用的 [****test-nginx****](https://github.com/openresty/test-nginx)  。
-一个测试用例 __.t__ 文件，通常用 \__DATA\__ 分割成 序言部分 和 数据部分。这里我们简单介绍下数据部分，
+针对功能，完善各种维度的测试用例，对插件做个全方位的测试吧！插件的测试用例，都在 **t/plugin** 目录下，可以前去了解。
+项目测试框架采用的 [\***\*test-nginx\*\***](https://github.com/openresty/test-nginx) 。
```
Comment (Yiyiyimu) [inline docs/zh/latest/plugin-develop.md:307]: remove duplicate stars

File: docs/zh/latest/plugins/openid-connect.md [hunk 5]
```diff
@@ -59,7 +62,7 @@ OAuth 2 / Open ID Connect（OIDC）插件为 APISIX 提供身份验证和自省
 ![token introspection](../../../assets/images/plugin/oauth-1.png)
 
 以下是 curl 命令，用于将插件启用到外部服务。
-通过自省请求标头中提供的令牌，此路由将保护 https://httpbin.org/get（echo 服务）。
+通过自省请求标头中提供的令牌，此路由将保护 https://httpbin.org/get (echo 服务）。
```
Comment (Yiyiyimu) [inline docs/zh/latest/plugins/openid-connect.md:65]: it should be a fullwidth bracket and a space, but not changed to halfwidth

**Review by Yiyiyimu — COMMENTED** [review]
Mostly LGTM

File: docs/en/latest/plugin-develop.md [hunk 6]
```diff
@@ -390,25 +394,25 @@ done
...
 
-Additionally, there are some convenience testing endpoints which can be found [here](https://github.com/apache/apisix/blob/master/t/lib/server.lua#L36). For example, see [proxy-rewrite](https://github.com/apache/apisix/blob/master/t/plugin/proxy-rewrite.lua). In test 42, the upstream `uri` is made to redirect `/test?new_uri=hello` to `/hello` (which always returns `hello world`). In test 43, the response body is confirmed to equal `hello world`, meaning the proxy-rewrite configuration added with test 42 worked correctly.
+Additionally, there are some convenience testing endpoints which can be found [here](https://github.com/apache/apisix/blob/master/t/lib/server.lua#L36). For example, see [proxy-rewrite](https://github.com/apache/apisix/blob/master/apisix/plugins/proxy-rewrite.lua). In test 42, the upstream `uri` is made to redirect `/test?new_uri=hello` to `/hello` (which always returns `hello world`). In test 43, the response body is confirmed to equal `hello world`, meaning the proxy-rewrite configuration added with test 42 worked correctly.
```
[the remark this answers is not in the document]
Comment (juzhiyuan) [reply docs/en/latest/plugin-develop.md:408]: oh!!

File: docs/zh/latest/FAQ.md (hunk 1 above)
Comment (juzhiyuan) [reply docs/zh/latest/FAQ.md]: updated

File: docs/zh/latest/admin-api.md (hunk 2 above)
Comment (juzhiyuan) [reply docs/zh/latest/admin-api.md]: just updated

File: docs/zh/latest/plugins/openid-connect.md (hunk 5 above)
Comment (juzhiyuan) [reply docs/zh/latest/plugins/openid-connect.md]: just updated

**Review by spacewander — APPROVED** [review]

**Review by Yiyiyimu — APPROVED** [review]

**Pull request merged**